### ISSUE TITLE
Support for Powershell Remoting over SSH to Windows targets

### ISSUE THREAD

**Issue by trondhindenes** [ansible/ansible#47706]
##### SUMMARY
Add support to allow using ssh (as opposed to http/winrm) as a communication protocol when targeting Windows targets

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
winrm

##### ADDITIONAL INFORMATION
Given that the target node has Powershell 6 installed, ssh is now an alternative communication channel to the existing http/winrm. This is documented here: https://docs.microsoft.com/en-us/powershell/scripting/core-powershell/ssh-remoting-in-powershell-core?view=powershell-6

It would be awesome to see support for this in Ansible. I'm not sure if the native ssh connection could be used, or if a new "psremoting_ssh" connection plugin is needed, I don't know enough about the core workings of Ansible connections to be able to supply more info. 

I would assume that this support would potentially drastically speed up configuration of windows nodes, since winrm is currently http/xml-based and very slow.

**Comment by ansibot** [comment]
Hi @trondhindenes, thank you for submitting this issue!

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
Files identified in the description:
* [lib/ansible/plugins/connection/winrm.py](https://github.com/ansible/ansible/blob/devel/lib/ansible/plugins/connection/winrm.py)

If these files are inaccurate, please update the `component name` section of the description or use the `!component` bot command.

[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by ansibot** [comment]
cc @jborean93 @nitzmahone
[click here for bot help](https://github.com/ansible/ansibullbot/blob/master/ISSUE_HELP.md)

**Comment by jborean93** [comment]
Hey @trondhindenes, having SSH connections to Windows hosts is something we wish to achieve but currently it requires some internal work to allow the current ssh connection to work with Windows hosts and PowerShell modules.

There are 2 ways we can achieve SSH with Windows;

* Use SSH directly and just execute `powershell.exe` like we currently do with the `winrm` connection plugin
* Use PSRP over SSH with the `psrp` connection plugin by setting `ansible_psrp_protocol: ssh`

The former will be the most useful as we can ensure compatibility with PSDesktop but requires some internal work before it can become a reality. I was really hoping to get this in for 2.7 but these changes proved to be too invasive at the time so have held off until we can do it correctly.

The latter is something I'm personally working on in my spare time but requires some extra features in a third party library. The main downside for this is that we are forced to use PSCore which in my testing is slower than PSDesktop to startup, not 100% compatible with our modules, and not actively tested in our test matrix.

> I would assume that this support would potentially drastically speed up configuration of windows nodes, since winrm is currently http/xml-based and very slow.

You will be surprised, the last time I tested SSH with Windows it took a similar time to execute modules as WInRM as most of the time taken here is due to the PowerShell interpreter being started for each task. The biggest benefit really for SSH and Windows is faster copy operations and easier authentication methods. Although I tested it about 9 months ago so things could have changed since then.

I'm going to close this as a duplicate and put your comment here https://github.com/ansible/ansible/issues/25344.
